sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -228,13 +288,17 @@ class StreamExecution(
     committedOffsets ++= availableOffsets
 
     // Check to see what new data is available.
-    val newData = uniqueSources.flatMap(s => s.getOffset.map(o => s -> o))
+    val newData = runUninterruptiblyInMicroBatchThread {
+      uniqueSources.flatMap(s => s.getOffset.map(o => s -> o))
+    }
```

Comment:
This is hard to understand (for future devs), why these should be uninterruptible. I think this needs more documentation.